Bugzilla/Auth/Persist/Cookie.pm

```diff
@@ -39,8 +39,11 @@ sub persist_login {
 
   my $ip_addr = remote_ip();
 
-  $dbh->do('INSERT INTO logincookies (cookie, userid, ipaddr, lastused)
-    VALUES (?, ?, ?, NOW())', undef, $login_cookie, $user->id, $ip_addr);
+  # Record the auth method the user used to get the new cookie
+  my $auth_type = ref $user->authorizer->{_info_getter}->{successful} || 'Unknown';
```

Comment:
You should be calling `$user->authorizer->successful_info_getter` instead of reaching into a private attribute.

Better yet to avoid repetition with `Bugzilla::Auth::Persist::Cookie`, add an `authentication_method` sub to `Bugzilla::User` which does the same.  Note `Bugzilla::Auth::Persist::Cookie` wraps its logic in `eval`, so it's worth doing that here too.

Reply:
Looking at the code it appears that `->{successful}` contains an object; it would be better to add a short name (eg. max 8 chars) to each `Bugzilla::Auth::Login` package and persist that.